integration-tests/cucumber-tests-platform/src/test/java/org/opensmartgridplatform/cucumber/platform/glue/steps/database/logging/DeviceLogItemSteps.java

```diff
@@ -44,11 +55,39 @@ public void iHaveADeviceLogItem(final Map<String, String> settings) {
         final boolean valid = true;
 
         final DeviceLogItem deviceLogItem = new DeviceLogItem.Builder().withDeviceIdentification(deviceIdentification)
-                .withDeviceUid(deviceUid).withDecodedMessage(decodedMessage).withEncodedMessage(encodedMessage)
-                .withIncoming(incoming).withOrganisationIdentification(organisationIdentification)
-                .withPayloadMessageSerializedSize(payloadMessageSerializedSize).withValid(valid).build();
+                .withDeviceUid(deviceUid)
+                .withDecodedMessage(decodedMessage)
+                .withEncodedMessage(encodedMessage)
+                .withIncoming(incoming)
+                .withOrganisationIdentification(organisationIdentification)
+                .withPayloadMessageSerializedSize(payloadMessageSerializedSize)
+                .withValid(valid)
+                .build();
+
+        final DeviceLogItem savedDeviceLogItem = this.deviceLogItemRepository.save(deviceLogItem);
+
+        final String modificationTimeString = getString(settings, PlatformKeys.KEY_MODIFICATION_TIME);
+        if (modificationTimeString != null) {
+            final ZonedDateTime modificationTime = ZonedDateTime.parse(modificationTimeString);
+            this.updateModificationTime(savedDeviceLogItem.getId(), modificationTime);
+        }
+    }
+
+    private void updateModificationTime(final Long deviceLogItemId, final ZonedDateTime modificationTime) {
+        if (modificationTime != null) {
+            final EntityManager em = this.txMgrLogging.getEntityManagerFactory().createEntityManager();
+            final EntityTransaction transaction = em.getTransaction();
 
-        this.deviceLogItemRepository.save(deviceLogItem);
+            transaction.begin();
+            final String sql = "UPDATE device_log_item SET modification_time = :modification_time WHERE id = :id";
+            final Query updateQuery = em.createNativeQuery(sql);
+            updateQuery.setParameter("id", deviceLogItemId);
+            updateQuery.setParameter("modification_time", modificationTime);
+            updateQuery.executeUpdate();
+            transaction.commit();
+
+            em.close();
```

Comment:
Why do we need a native query?
I don't think a native query should be in a steps class.

Reply:
How can I set the modificationTime in a cleaner way? It is normally set automatically, when an item is saved. For these test cases, I need a specific modificationTime.

Reply:
Ok, it is clear to me why you do need a native query, It would be cleaner to make it a method in a repository class.